File: dih_side_error.c
#include<stdio.h>
#include<math.h>

double error(double *x){
	double sum,a,n,dif;
	int i;

	for(i = 0;i < 20;i++){
		sum += x[i];
	}
	a = sum / 20.0;

	n = 0;
	for(i = 0;i < 20;i++){
		n += pow(x[i] - a,2.0);
	}

	dif = n / 19.0;
	return sqrt(dif);
}

main(){
	FILE *fp;
	char *fname = "dih_side.txt";
	double n[10000],s[10000];
	double sum_syn,sum_anti;
	double sam_syn,sam_anti;
	double ave_syn,ave_anti;
	double error_syn,error_anti;
	double x[50],y[50];
	int i,j,p;

	fp = fopen(fname,"r");
	if(fp == NULL){
		printf("%sファイルが開けません。\n",fname);
		return -1;
	}

	i = 0;
	while((fscanf(fp,"%lf %lf",&n[i],&s[i])) != EOF){
		i++;
	}
	fclose(fp);

	j = 0;
	for(i = 0;i < 20;i++){
		sam_syn = 0;
		sam_anti = 0;
		for(;j < p + 500;j++){
			if(-90.00 < s[j] && s[j] < 90.00){
				sam_syn++;
			}else if(-180.00 < s[j] && s[j] < -90.00){
				sam_anti++;
			}else if(90.00 < s[j] && s[j] < 180.00){
				sam_anti++;
			}
		}
		sum_syn += sam_syn;
		sum_anti += sam_anti;
		x[i] = sam_syn / 500.0;
		y[i] = sam_anti / 500.0;
		p = j;
	}

	ave_syn = sum_syn / 10000.0;
	ave_anti = sum_anti / 10000.0;

	error_syn = error(x);
	error_anti = error(y);

	printf("ave_syn = %lf +-%lf\nave_anti = %lf +-%lf\n",ave_syn,error_syn,ave_anti,error_anti);

	return 0;
}

File: dih_side_error.py
import math
def error(n):
    dx = 0.0
    for i in range(20):
        dx += n[i]
    dx = dx / 20.0
    sum = 0.0
    for i in range(20):
        sum += ((n[i] - dx) ** 2)

    dif = sum / 19.0
    return math.sqrt(dif)

sum_syn = 0
sum_anti = 0
sam_syn = 0
sam_anti = 0
n = 20
x = n*[0]
y = n*[0]
i = 0
j = 0
fname = "dih_side.txt"
for f in open(fname,"r"):
    data = f.split()
    dih = float(data[1])
    if -90.0 < dih < 90.0:
        sum_syn += 1
        sam_syn += 1
    else:
        sum_anti += 1
        sam_anti += 1
    i += 1
    if i == 500:
        x[j] = sam_syn / 500.0
        y[j] = sam_anti / 500.0
        j += 1
        sam_syn = 0
        sam_anti = 0
        i = 0
for i in range(20):
    print "x[%d] = %lf y[%d] = %lf" % (i,x[i],i,y[i])

ave_syn = sum_syn / 10000.0
ave_anti = sum_anti / 10000.0

print "ave_syn = %lf  +-%lf" % (ave_syn,error(x))
print "ave_anti= %lf  +-%lf" % (ave_anti,error(y))
